targets/psa_switch/psa_switch.cpp

```diff
@@ -308,55 +308,56 @@ PsaSwitch::ingress_thread() {
     input_buffer.pop_back(&packet);
     if (packet == nullptr) break;
 
-    Parser *parser = this->get_parser("ingress_parser");
-    Pipeline *ingress_mau = this->get_pipeline("ingress");
-    phv = packet->get_phv();
-
     port_t ingress_port = packet->get_ingress_port();
     (void) ingress_port;
     BMLOG_DEBUG_PKT(*packet, "Processing packet received on port {}",
                     ingress_port);
 
+    phv = packet->get_phv();
+
+    Parser *parser = this->get_parser("ingress_parser");
     parser->parse(packet.get());
+
+    // set default std metadata values
+    phv->get_field("psa_ingress_output_metadata.drop").set(1);
+
+    Pipeline *ingress_mau = this->get_pipeline("ingress");
     ingress_mau->apply(packet.get());
+    packet->reset_exit();
 
     // prioritize dropping if marked as such - do not move below other checks
-    if (phv->has_field("psa_ingress_output_metadata.drop")) {
-      Field &f_drop = phv->get_field("psa_ingress_output_metadata.drop");
-      if (f_drop.get_int()) {
-        continue;
-      }
-    }
-
-    // handling multicast
-    unsigned int mgid = 0u;
-    if (phv->has_field("psa_ingress_output_metadata.multicast_group")) {
-      Field &f_mgid = phv->get_field("psa_ingress_output_metadata.multicast_group");
-      mgid = f_mgid.get_uint();
-
-      if(mgid != 0){
-        const auto pre_out = pre->replicate({mgid});
-        auto packet_size = packet->get_register(PACKET_LENGTH_REG_IDX);
-        for(const auto &out : pre_out){
-          auto egress_port = out.egress_port;
-          std::unique_ptr<Packet> packet_copy = packet->clone_with_phv_ptr();
-          packet_copy->set_register(PACKET_LENGTH_REG_IDX, packet_size);
-          enqueue(egress_port, std::move(packet_copy));
-        }
-        continue;
-      }
+    Field &f_drop = phv->get_field("psa_ingress_output_metadata.drop");
+    if (f_drop.get_int()) {
+      BMLOG_DEBUG_PKT(*packet, "Dropping packet marked as such from ingress");
+      continue;
     }
 
-    packet->reset_exit();
     Field &f_egress_spec = phv->get_field("psa_ingress_output_metadata.egress_port");
     port_t egress_spec = f_egress_spec.get_uint();
     port_t egress_port = egress_spec;
-
     egress_port = egress_spec;
     BMLOG_DEBUG_PKT(*packet, "Egress port is {}", egress_port);
```

Comment:
The 5 lines of code above starting with get_field on egress_port, down to loggin the egress_port, are all specific to unicast packets, and should probably be done later after the mgid != 0 check is false, just before the enqueue call.

Reply:
Fixed! Thanks for pointing this out.